std/http/server.ts

```diff
@@ -113,11 +113,58 @@ export class ServerRequest {
   }
 }
 
-export class Server implements AsyncIterable<ServerRequest> {
+export type ServerErrorEventListenerOrEventListenerObject =
+  | ServerErrorListener
+  | ServerErrorListenerObject;
+
+export interface ServerErrorListener {
+  (evt: ServerErrorEvent): void | Promise<void>;
+}
+
+export interface ServerErrorListenerObject {
+  handleEvent(evt: ServerErrorEvent): void | Promise<void>;
+}
+
+export interface ServerErrorEventInit extends ErrorEventInit {
+  server: Server;
+  origin: ServerErrorEvent["origin"];
+  connection?: Deno.Conn;
+}
+
+export class ServerErrorEvent extends ErrorEvent {
+  server: Server;
+  origin: "listener" | "request";
```

Comment:
Maybe we should rename these (origin, listener, request)... My English is not good enough.